lib/rules/no-array-index-key.js

```diff
@@ -21,9 +21,10 @@ const messages = {
 
 module.exports = {
   meta: {
+    type: 'problem',
     docs: {
       description: 'Prevent usage of Array index in keys',
-      category: 'Best Practices',
+      category: 'Possible Errors',
```

Comment:
changing the category is a breaking change; also, this is indeed a best practice, and does not necessarily indicate a problem.